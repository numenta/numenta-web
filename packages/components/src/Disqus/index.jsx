// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'
import root from 'window-or-global'

import styles from './index.css'


/**
 * Disqus Comments Thread React Component.
 *  Custom, as `react-disqus-thread` is not working.
 * @see github:mzabriskie/react-disqus-thread/issues/14#issuecomment-234407160
 */
class Disqus extends React.Component {

  constructor(props) {
    super(props)
    this.state = props
  }

  componentWillMount() {
    const {document} = root
    const component = this
    const script = document.createElement('script')
    root.disqus_config = function () {
      this.page.category_id = component.state.category_id
      this.page.identifier = component.state.identifier
      this.page.title = component.state.title
      this.page.url = component.state.url
      this.callbacks.onNewComment = component.state.onNewComment
    }
    script.src = `//${this.state.shortname}.disqus.com/embed.js`
    script.async = true
    document.body.appendChild(script)
  }

  componentWillReceiveProps(nextProps) {
    this.setState(nextProps)
  }

  render() {
    return (
      <div className={styles.disqus} id="disqus_thread" />
    )
  }

}

Disqus.propTypes = {
  category_id: React.PropTypes.string,
  identifier: React.PropTypes.string,
  onNewComment: React.PropTypes.func,
  shortname: React.PropTypes.string.isRequired,
  title: React.PropTypes.string,
  url: React.PropTypes.string,
}

export default Disqus
