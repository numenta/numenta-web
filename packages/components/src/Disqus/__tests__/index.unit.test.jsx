import React from 'react'
import renderer from 'react-test-renderer'

import Disqus from '../../Disqus'


describe('Disqus React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Disqus
        category_id="abcd1234"
        identifier="unique"
        onNewComment={() => {}}
        shortname="disqus"
        title="Title"
        url="http://disqus.com"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
