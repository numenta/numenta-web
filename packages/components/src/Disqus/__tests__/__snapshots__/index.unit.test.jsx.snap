exports[`Disqus React component Renders correctly 1`] = `
<div
  className="disqus"
  id="disqus_thread" />
`;
